medical_pathology_import/wizards/medical_pathology_import.py

```diff
@@ -0,0 +1,165 @@
+    """ It provides a model for importing ICD-10-CM codes to Odoo. """
+
+    _name = 'medical.pathology.import'
+    _description = 'Medical Pathology Import'
+
+    zip_uri = fields.Char(
+        help='URI to full ZIP file',
+        string='ZIP File URI',
+    )
+    file_name = fields.Char(
+        help='Name of file (inside of ZIP) representing full tabular XML',
+        string='XML File Name',
+    )
+    code_type_id = fields.Many2one(
+        string='Code Type',
+        comodel_name='medical.pathology.code.type',
+        required=True,
+        help='Identifies the code type that is being imported. If you are not'
+             ' sure what to choose, select an importer first and leave the'
+             ' resulting default',
+    )
+    importer_type = fields.Selection(
+        selection=lambda s: s._get_importer_types(),
+        required=True,
+        string='Importer Type',
+    )
+
+    @api.model
+    def _get_importer_types(self):
+        return []
+
+    @api.multi
+    @api.onchange('importer_type')
+    def _onchange_importer_type(self):
+        """ It calls gets default data by importer type.
+
+        To plug into this interface, importers should implement method
+        ``_onchange_importer_type_*``, such as
+        ``_onchange_importer_type_icd10``.
+        """
+        try:
+            method = getattr(
+                self, '_onchange_importer_type_%s' % self.importer_type,
+            )
+            method()
+        except AttributeError:
+            self.zip_uri = False
+            self.file_name = False
+
+    @api.model
+    def create(self, vals):
+        """ It begins import process after create """
+        record = super(MedicalPathologyImport, self).create(vals)
+        record.do_import()
+        return record
+
+    @api.multi
+    def do_import(self):
+        """ It calls the import method for the type of importer.
+
+        Import methods should be named ``do_import_*``, such as
+        ``do_import_icd10`` or ``do_import_snomed``.
+        """
+        self.ensure_one()
+        method = getattr(self, 'do_import_%s' % self.importer_type)
+        method()
+
+    @api.multi
+    def _get_pathology_xml_id(self, code):
+        """ It returns the XML ID for a pathology code """
+        return 'medical_pathology_%s' % code.replace('.', '_')
+
+    @api.multi
+    def _get_pathology_category_xml_id(self, name):
+        """ IT returns the XML ID for a pathology category by name """
+        name = name.replace('.', '_').replace('-', '_')
+        return 'medical_pathology_category_%s' % name
+
+    @api.multi
+    def _upsert_pathology(
+        self, name, code, category, code_type, id_pref, parent=None, note=None,
+    ):
+        """ It updates or creates a new Pathology for arguments.
+
+        Args:
+            name: (str) Name of pathology to create.
+            code: (str) Code for pathology.
+            category: (medical.pathology.category) Singleton representing the
```

Comment:
Models should be the actual name and not the `_name`. So `MedicalPathologyCategory`